src/promptflow-recording/promptflow/recording/azure/utils.py

```diff
@@ -140,6 +163,57 @@ def sanitize_upload_hash(value: str) -> str:
         value,
         flags=re.IGNORECASE,
     )
+    value = sanitize_pf_run_ids(value)
+    return value
+
+
+def sanitize_pf_run_ids(value: str) -> str:
```

Comment:
just curious, why we need such a utility here? is there any limitation that [`randstr`](https://github.com/microsoft/promptflow/blob/e16aa033cc88e0cb671d17eb1c3a2b8d403aee81/src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py#L332-L345) cannot support the scenario?

Reply:
The problem is that run name is generated from prefix+randstr+timestemp, so datetime also needs to be mocked. I wanted to minimize number of mocks inside the e2e test.

Reply:
IIRC, run name can be override, so that we don't need to handle timestamp thing?

Reply:
however, this utility looks good to me; I will take a look on the scenario you mention.